pkg/verifier/trust/verifier.go

```diff
@@ -80,12 +80,21 @@ func (v *Verifier) VerifyByPolicy(namespace string, img *image.Reference, creden
 		}
 	}
 
+	authEndpoint, err := v.trust.CheckAuthRequired(notaryURL, img)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Deny %q, could not resolve the auth-endpoint, %s", img.String(), err.Error())
+	}
+
 	credentials = append(credentials, credential.Credential{})
```

Comment:
The append of blank credentials here is an alternative and poor way to solve part of the problem addressed by this PR. It relies on the auth service handing out a free token (for empty creds) when no authentication is needed, given the change proposed we know that no token is needed because authEndpoint is nil. 
My suggestion does not lead to the prettiest code, there may be a better way.